src/Bicep.Core/Analyzers/Linter/Rules/MaxNumberResourcesRule.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Semantics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class MaxNumberResourcesRule : LinterRuleBase
+    {
+        public new const string Code = "max-resources";
+        public MaxNumberResourcesRule() : base(
+            code: Code,
+            description: CoreResources.MaxNumberResourcesRuleDescription,
+            docUri: new Uri($"https://aka.ms/bicep/linter/{Code}"),
+            diagnosticLabel: Diagnostics.DiagnosticLabel.Unnecessary,
+            diagnosticLevel: DiagnosticLevel.Error)
+        { }
+
+        public override string FormatMessage(params object[] values)
+        {
+            return string.Format(CoreResources.MaxNumberResourcesRuleMessageFormat, values);
+        }
+
+        override public IEnumerable<IDiagnostic> AnalyzeInternal(SemanticModel model)
+        {
+            if (model.Root.ResourceDeclarations.Count() > 800) {
```

Comment:
>ResourceDeclarations [](http://example.com/codeflow?start=27&length=20)

@anthony-c-martin do you remember if nested resources count against the limit? I would assume they do.

Reply:
This issue doesn't apply to vars, outputs, or params because those can only exist at the top level.

Reply:
I've did a small test and nested resources are counted towards the limit. I'll change the rule to take that into account.